packages/editor/src/utils/media-upload/index.js

```diff
@@ -35,13 +35,13 @@ export default function mediaUpload( {
 	const wpAllowedMimeTypes = getEditorSettings().allowedMimeTypes;
 	maxUploadFileSize =
 		maxUploadFileSize || getEditorSettings().maxUploadFileSize;
-
+	const currentPostId = getCurrentPostId();
 	uploadMedia( {
 		allowedTypes,
 		filesList,
 		onFileChange,
 		additionalData: {
-			post: getCurrentPostId(),
+			...( ! isNaN( currentPostId ) ? { post: currentPostId } : {} ),
```

Comment:
Nitpick, but I'd find it more readable if it were something along the lines of:
```js
const postData = isNaN( currentPostId ) ? {} : { post: currentPostId };
// ...
additionalData: {
    ...postData,
    ...additionalData
};
```

Reply:
In the case of templates, the `isNan` test will return `false`. Is that a problem?

`! isNaN('twentytwentyfour//home') === false // true`